tests/test_domain_js.py

```diff
@@ -120,21 +120,28 @@ def find_obj(mod_name, prefix, obj_name, obj_type, searchmode=0):
 
     assert (find_obj(None, None, 'NONEXISTANT', 'class') == (None, None))
     assert (find_obj(None, None, 'NestedParentA', 'class') ==
-            ('NestedParentA', ('roles', 'class')))
+            ('NestedParentA', ('roles', 'nestedparenta', 'class')))
     assert (find_obj(None, None, 'NestedParentA.NestedChildA', 'class') ==
-            ('NestedParentA.NestedChildA', ('roles', 'class')))
+            ('NestedParentA.NestedChildA',
+             ('roles', 'nestedparenta-nestedchilda', 'class')))
```

Comment:
This is good example for new_id. It is regulated to small case and invalid characters are converted to hyphens.